### PR TITLE
feat: [CDS-76027]: remove dependency on time_bucket_gapfill

### PR DESCRIPTION
**Pull request by RishabhGupta34**
## Describe your changes
Remove dependency on time bucket gapfill

Tested locally:
Deployment data for last 30/60/90/1 year, and compared it with the older graph(without these changes)

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52110)

### CODE REVIEW

**Comment by RishabhGupta34** [comment]
trigger uts

**Comment by RishabhGupta34** [comment]
trigger uts

**Comment by RishabhGupta34** [comment]
trigger commonchecks, uts

**Comment by RishabhGupta34** [comment]
trigger commonchecks

**Comment by RishabhGupta34** [comment]
trigger uts

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 6 of 7 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52110)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52110)* status: 11 file(s) to review. 1 of 2 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java` line 315 at r2](https://reviewable.io/reviews/harness/harness-core/52110#-Nd28aB_88LGMXSxY6In:-Nd28aBa5OL7pAhth1S-:b-bbno7c) ([raw file](https://github.com/harness/harness-core/blob/a41fb6dc573ba7d237b80fadd6f6146cb51788cf/125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java#L315)):*
> ```Java
>         Field<Long> t3_startts = DSL.field("t3.startts", Long.class);
>         Field<String> t3_status = DSL.field("t3.status", String.class);
>         Field<Integer> caseConditionStep = DSL.case_().when(t3_status.isNotNull(), DSL.count()).otherwise(0);
> ```

@harness-david We will be making changes behind a flag to control issues we may observe if any

**Comment by RishabhGupta34** [comment]
trigger uts

**Comment by RishabhGupta34** [comment]
trigger ut2, ut4

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52110)* status: 11 file(s) to review. 1 of 2 approvals obtained.

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52110)* status: 11 file(s) to review. 1 of 2 approvals obtained.

File: 125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java [hunk 1]
```diff
@@ -300,10 +306,39 @@ public List<PipelineExecutionSummaryCd> getPipelineExecutionsForGivenExecutionSt
...
+                                                    .and(PIPELINE_EXECUTION_SUMMARY_CD.PROJECTIDENTIFIER.eq(
+                                                        (Field<String>) orgProjectTable.field(PROJECT_ID)))))
+                          .asTable("t3"))
+            .on(tb_status_status.eq(t3_status))
+            .and(tb_status_startts.eq(t3_startts.subtract(t3_startts.mod(groupBy.getNoOfMilliseconds()))))
+            .groupBy(tb_status_startts, t3_status, tb_status_status)
+            .orderBy(tb_status_startts)
+            .fetchInto(TimeWiseExecutionSummary.class);
+      }
+      Field<Long> epoch = DSL.field("time_bucket_gapfill(" + groupBy.getNoOfMilliseconds() + ", {0})", Long.class,
```
Comment (harness-david) [inline 125-cd-nextgen/src/main/java/io/harness/cd/TimeScaleDAL.java:340]: Why is time_bucket_gapfill being used?  The goal is to remove all traces of it.

**Review by harness-david — APPROVED** [review]
The changes look accurate to me.  The CDS_REMOVE_TIME_BUCKET_GAPFILL_QUERY will need to be enabled right away though.

**Comment by RishabhGupta34** [comment]
yeah, we will start enabling it for some customers, from next week itself

**Comment by RishabhGupta34** [comment]
trigger review

**Comment by RishabhGupta34** [comment]
trigger review

**Pull request merged**